cmake/sysbuild/suit.cmake

```diff
@@ -164,14 +166,35 @@ function(suit_create_package)
   set(CORE_ARGS)
   set(STORAGE_BOOT_ARGS)
 
-  sysbuild_get(CONFIG_SUIT_ENVELOPE_EDITABLE_TEMPLATES_LOCATION IMAGE ${DEFAULT_IMAGE} VAR CONFIG_SUIT_ENVELOPE_EDITABLE_TEMPLATES_LOCATION KCONFIG)
-  suit_set_absolute_or_relative_path(${CONFIG_SUIT_ENVELOPE_EDITABLE_TEMPLATES_LOCATION} ${PROJECT_BINARY_DIR} INPUT_TEMPLATES_DIRECTORY)
-  sysbuild_get(ENVELOPE_SHALL_BE_SIGNED IMAGE ${DEFAULT_IMAGE} VAR CONFIG_SUIT_ENVELOPE_SIGN KCONFIG)
+  suit_set_absolute_or_relative_path(${SB_CONFIG_SUIT_ENVELOPE_EDITABLE_TEMPLATES_LOCATION} ${PROJECT_BINARY_DIR} INPUT_TEMPLATES_DIRECTORY)
+  set(ENVELOPE_SHALL_BE_SIGNED ${SB_CONFIG_SUIT_ENVELOPE_SIGN})
   if(NOT DEFINED ENVELOPE_SHALL_BE_SIGNED)
     set(ENVELOPE_SHALL_BE_SIGNED FALSE)
   endif()
 
+  list(APPEND CORE_ARGS
+    --core sysbuild,,,${CMAKE_BINARY_DIR}/zephyr/.config
+  )
+
+  foreach(image ${IMAGES})
+    sysbuild_get(BINARY_DIR IMAGE ${image} VAR APPLICATION_BINARY_DIR CACHE)
+    sysbuild_get(BINARY_FILE IMAGE ${image} VAR CONFIG_KERNEL_BIN_NAME KCONFIG)
+
+    set(BINARY_FILE "${BINARY_FILE}.bin")
+
+    list(APPEND CORE_ARGS
+      --core ${image},${SUIT_ROOT_DIRECTORY}${image}.bin,${BINARY_DIR}/zephyr/edt.pickle,${BINARY_DIR}/zephyr/.config
```

Comment:
This code looks like we create the `CORE_ARGS` twice for each core... what is the catch here?

Reply:
I have talked with @robertstypa about this a while back and my idea was that it should be possible to reference any image within the sysbuild by using its name from `ExternalZephyrProject_Add`. In case of the main application, it can be referenced through `application` or through the sample name, such as `smp_transfer`. Both names point to the same binary and have the same data.